src/components/UsedByItem.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { FC } from "react";
+import ResourceLink from "./ResourceLink";
+import { LxdUsedBy } from "util/usedBy";
+import { ResourceIconType } from "./ResourceIcon";
+import { useQuery } from "@tanstack/react-query";
+import { fetchImage } from "api/images";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  item: LxdUsedBy;
+  project: string;
+  type: ResourceIconType;
+  to: string;
+}
+
+const UsedByItem: FC<Props> = ({ item, project, type, to }) => {
+  const { data: image } = useQuery({
+    queryKey: [queryKeys.images, item.name],
+    queryFn: () => fetchImage(item.name, project),
+    enabled: type === "image",
+  });
+
+  let label = item.name;
+  if (type === "image") {
+    label = image?.properties?.description || item.name;
+  }
+
+  if (type === "snapshot") {
+    if (item.instance) {
+      label = `${item.instance} ${item.name}`;
+    }
+
+    if (item.volume) {
+      label = `${item.volume} ${item.name}`;
+    }
```

Comment:
I think we should follow the same paradigm as with the profile: show the instance (or volume) as a separate chip with a / next to it, to show the snapshot is a nested object.

Reply:
@edlerd do you think there is a bug with snapshots in used by? I can't seem to figure out how to get snapshots to showup in the `used_by` attributes for instances and volumes?

Reply:
Not sure if I understand the question. If you ask why the snapshots don't show up in the used by section of an instance or volume -- I think they shouldn't.

Reply:
So tried with a `dir` storage pool as suggested, still not seeing any snapshot (I have a snapshot created for the instance and  custom volume on the pool shown below). You mentioned snapshots shows up for you in this scenario right?

![Screenshot from 2024-10-17 15-56-05](https://github.com/user-attachments/assets/2178f7e1-86d0-4cac-8109-e88c0f97e3a0)

Reply:
Yes they do, using latest/stable and your branch here:

![image](https://github.com/user-attachments/assets/b1e28fbb-9efb-4464-99b0-a609915bcf98)

Reply:
I don't understand..., it's not in the API response for me, I'm using latest/edge though, so maybe I will try latest/stable

Reply:
In any case, I don't think this issue should be a blocker for this PR, but I will play around with this a bit more outside of the discussion here as well

Reply:
Can confirm on latest/edge the snapshots are missing. Please report this as a bug upstream.

Reply:
Will do! Thanks for checking :)

Reply:
keeping this comment open until the issue is created

Reply:
[reported](https://github.com/canonical/lxd/issues/14291)